diff --git a/src/base/SkTSearch.cpp b/src/base/SkTSearch.cpp
index d91772e03b..9b870635cf 100644
--- a/src/base/SkTSearch.cpp
+++ b/src/base/SkTSearch.cpp
@@ -55,7 +55,7 @@ int SkStrSearch(const char*const* base, int count, const char target[],
     return hi;
 }
 
-int SkStrSearch(const char*const* base, int count, const char target[],
+SK_API int SkStrSearch(const char*const* base, int count, const char target[],
                 size_t elemSize)
 {
     return SkStrSearch(base, count, target, strlen(target), elemSize);
diff --git a/src/core/SkTextBlobPriv.h b/src/core/SkTextBlobPriv.h
index 6a4c1531cf..85005527c5 100644
--- a/src/core/SkTextBlobPriv.h
+++ b/src/core/SkTextBlobPriv.h
@@ -182,7 +182,7 @@ private:
  *         .....
  *    }
  */
-class SkTextBlobRunIterator {
+class SK_API SkTextBlobRunIterator {
 public:
     SkTextBlobRunIterator(const SkTextBlob* blob);
 
diff --git a/src/ports/SkDebug_stdio.cpp b/src/ports/SkDebug_stdio.cpp
index 78c7072bd0..3916fed46b 100644
--- a/src/ports/SkDebug_stdio.cpp
+++ b/src/ports/SkDebug_stdio.cpp
@@ -7,13 +7,14 @@
 
 #include "include/private/base/SkFeatures.h"
 #include "include/private/base/SkLoadUserConfig.h"
+#include "include/private/base/SkAPI.h"
 
 #if !defined(SK_BUILD_FOR_WIN) && !defined(SK_BUILD_FOR_ANDROID)
 
 #include <stdarg.h>
 #include <stdio.h>
 
-void SkDebugf(const char format[], ...) {
+SK_API void SkDebugf(const char format[], ...) {
     va_list args;
     va_start(args, format);
 #pragma GCC diagnostic push
diff --git a/src/utils/SkOSPath.h b/src/utils/SkOSPath.h
index 783e985fd6..b95cf7cb95 100644
--- a/src/utils/SkOSPath.h
+++ b/src/utils/SkOSPath.h
@@ -13,7 +13,7 @@
 /**
  *  Functions for modifying SkStrings which represent paths on the filesystem.
  */
-class SkOSPath {
+class SK_API SkOSPath {
 public:
 #ifdef _WIN32
     static constexpr char SEPARATOR = '\\';
diff --git a/src/xml/SkDOM.h b/src/xml/SkDOM.h
index bceb8234e9..70b60ddb77 100644
--- a/src/xml/SkDOM.h
+++ b/src/xml/SkDOM.h
@@ -21,7 +21,7 @@ class SkDOMParser;
 class SkStream;
 class SkXMLParser;
 
-class SkDOM : public SkNoncopyable {
+class SK_API SkDOM : public SkNoncopyable {
 public:
     SkDOM();
     ~SkDOM();
